### ISSUE TITLE
List items + unnecessary line breaks; does not handle sentences with multiple words

### ISSUE THREAD

**Issue by bastones** [andrerpena/react-mde#244]
**Issue 1:**

I have noticed when using a toolbar command that triggers `makeList()` (such as unordered lists), an unnecessary extra line is added each time, caused by:

```
var breaksBefore = Array(breaksBeforeCount + 1).join("\n");

...

var breaksAfter = Array(breaksAfterCount + 1).join("\n");
```

**See:** lib/js/commands/listCommands.js

As soon as I remove `+ 1` from both variables, the line breaks are added after the current line as expected.

I realise GitHub has this strange behaviour as well, but it's just annoying and unexpected. The editor shouldn't add extra lines unecessarily when the user does not expect it.

**Issue 2:**

When you have more than one word on a single line and press any toolbar command that triggers `makeList()`, it takes the last word from that sentence and moves it to a new line? Why? This is completely unexpected behaviour and I have no idea why even GitHub does this either.

It should just create a new line with `- `, ready for the user to begin typing their next list item. Only if a user highlights some words from the sentence should it attempt to take those words and move them to a new line, as it does currently.

Please fix these issues :-(.

**Comment by S-Nuttapong** [comment]
- askdlaksd
- asdlkjasd

- asdlmkaskdm

**Comment by plamenkoyovchev** [comment]
This issue is closed as completed but it is reproducible in 11.5.0  ?
